base_layer/core/src/proof_of_work/lwma_diff.rs

```diff
@@ -109,7 +110,7 @@ impl LinearWeightedMovingAverage {
         // k is the sum of weights (1+2+..+n) * target_time
         let k = n * (n + 1) * self.target_time / 2;
         #[allow(clippy::cast_possible_truncation)]
-        let target = (ave_difficulty * k / weighted_times) as u64;
+        let target = (ave_difficulty * k / weighted_times).to_u64().unwrap_or(u64::MAX);
```

Comment:
You can remove the `#[allow(clippy::cast_possible_truncation)]`

Reply:
We can use stdlibs conversion without using the num-traits dependency
```suggestion
        let target = u64::try_from(ave_difficulty * k / weighted_times).unwrap_or(u64::MAX);
```